### PR TITLE
Create mlcompass_maxtext_gke dag

### PR DESCRIPTION
**Pull request by ortibazar**
# Description

Add MLCompass MaxText GKE runner for deploying and executing MaxText models, and add a `skip_prost_process` parameter to the xlml/apis/task to allow workflows bypass post-processing tasks.

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
```
  blaze build -c opt third_party/mlcompass/executors/runners:tf_runner
  blaze-bin/third_party/mlcompass/executors/runners/tf_runner \
    --config_path=xlml/xlml_dev.pbtxt \
    --benchmark_ids=1 \
    --xlml_environment_name=orti-test \
    --xlml_dag_name=mlcompass_maxtext_gke \
    --xlml_build_docker_image=true \
    --alsologtostderr
```

**List links for your tests (use go/shortn-gen for any internal link):** ...
https://a2ca3cdf681d42a3aceb01b7d66a6681-dot-us-central1.composer.googleusercontent.com/dags/mlcompass_maxtext_gke/grid?dag_run_id=manual__2024-11-21T18%3A04%3A51%2B00%3A00

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by brajiang — APPROVED** [review]

**Review by mbzomowski — COMMENTED** [review]
Please add a description for this PR

File: xlml/apis/task.py [hunk 1]
```diff
@@ -187,7 +188,8 @@ def run(
       run_model, gcs_path = self.run_model(
           gcs_location, use_vertex_tensorboard, use_pathways
       )
-      run_model >> self.post_process(gcs_path)
+      if not skip_post_process:
```
[the remark this answers is not in the document]
Comment (ortibazar) [reply xlml/apis/task.py:191]: Even though test fails, the overall workflow does not fail.

Example: http://screen/85xM8btN5PyPaVd

After skipping post process: http://screen/AfcuFpwRqnY6Gi6

**Comment by ortibazar** [comment]
> Please add a description for this PR

Done

File: .github/requirements.txt [hunk 2]
```diff
@@ -9,3 +9,4 @@ jsonlines
 tensorflow-cpu
 kubernetes
 pyarrow
+apache-airflow-providers-google
```
Comment (RissyRan) [inline .github/requirements.txt:12]: is this needed for github runners?

Comment (RissyRan) [reply .github/requirements.txt:12]: discussed offline

Comment (ortibazar) [reply .github/requirements.txt:12]: Yes, otherwise some tests fail with dependency error

**Review by RissyRan — APPROVED** [review]

**Pull request merged**